frontend/src/components/MapView/LeftPanel/layersPanel/MenuSwitch/SwitchItem/index.tsx

```diff
@@ -79,16 +83,9 @@ const SwitchItem = memo(({ classes, layer, extent }: SwitchItemProps) => {
 
   const selectedActiveLayer = useMemo(() => {
     return selected
-      ? selectedLayers.filter(sl => {
-          return (
-            (group?.activateAll &&
-              group?.layers.find(l => l.id === sl.id && l.main)) ||
-            (!group?.activateAll &&
-              group?.layers.map(l => l.id).includes(sl.id))
-          );
-        })
+      ? selectedLayers.filter(sl => filterActiveLayers(sl, sl))
```

Comment:
Maybe add a comment? It feels weird to have a function that takes twice the exact same argument

Reply:
Of course

Reply:
@HarryMytilinaios if the change was useful you can keep it, I would just recommend adding a comment to explicit the behavior

Reply:
I think this was not the case here and the change was not correct this is why I removed it.